### PR TITLE
update docstrings for df batch4

### PR DESCRIPTION
**Pull request by mfonekpo**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [x] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/dataframe.py [hunk 1]
```diff
@@ -890,28 +890,57 @@ def drop(self, *columns: str | Iterable[str]) -> Self:
             *columns: Names of the columns that should be removed from the dataframe.
 
         Examples:
-            >>> import pandas as pd
```
Comment (MarcoGorelli) [inline narwhals/dataframe.py:893]: this docstring's fine it doesn't need updating

Comment (mfonekpo) [reply narwhals/dataframe.py:893]: Can you give me a little more detail, please? I changed a lot of lines, Should I revert everything to the way it was? And please note, in line 717, you requested for it to be changed from `nw.DataFrame()` to `nw.from_native()`

File: narwhals/dataframe.py [hunk 2]
```diff
@@ -883,35 +921,33 @@ def head(self, n: int) -> Self:
...
-        Remove columns from the dataframe.
+        r"""
+        Remove columns from the LazyFrame.
 
         Arguments:
-            *columns: Names of the columns that should be removed from the dataframe.
+            *columns: Names of the columns that should be removed from the
+                      dataframe. Accepts column selector input.
 
         Examples:
```
Comment (MarcoGorelli) [inline narwhals/dataframe.py:931]: the `select` changes are fine, but the `drop` ones need reverting

File: narwhals/dataframe.py [hunk 3]
```diff
@@ -884,34 +922,35 @@ def head(self, n: int) -> Self:
 
     def drop(self, *columns: str | Iterable[str]) -> Self:
         """
-        Remove columns from the dataframe.
+        Remove columns from the LazyFrame.
+
+        This is a test line to check if changes are actually being pushed.
```
Comment (MarcoGorelli) [inline narwhals/dataframe.py:928]: 😄 it's definitely being pushed

but you see here that it shows a lot of changes for the `drop` method? there shouldn't be any changes here

File: narwhals/dataframe.py [hunk 4]
```diff
@@ -884,7 +922,7 @@ def head(self, n: int) -> Self:
 
     def drop(self, *columns: str | Iterable[str]) -> Self:
         """
-        Remove columns from the dataframe.
+           Remove columns from the dataframe.
```
Comment (MarcoGorelli) [inline narwhals/dataframe.py:925]:
```suggestion
       Remove columns from the dataframe.
```

**Comment by MarcoGorelli** [comment]
thanks

there's still some things, but i'll push a commit to this branch, don't worry about it, feel free to start a different task

**Comment by MarcoGorelli** [comment]
closing then, as this has been taken care of in other prs - thanks for your pr! going strong on the left join one 💪

**Pull request closed without merging**